src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/mapper/induction/PreviousQualificationsEntityMapper.kt

```diff
@@ -2,33 +2,101 @@ package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.ma
+import org.springframework.beans.factory.annotation.Autowired
 import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.entity.induction.PreviousQualificationsEntity
 import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.entity.induction.QualificationEntity
 import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.mapper.ExcludeJpaManagedFields
 import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.mapper.ExcludeJpaManagedFieldsIncludingDisplayNameFields
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.mapper.ExcludeReferenceField
 import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.mapper.GenerateNewReference
 import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.induction.PreviousQualifications
 import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.induction.Qualification
 import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.induction.dto.CreatePreviousQualificationsDto
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.induction.dto.UpdatePreviousQualificationsDto
 
 @Mapper(
   uses = [
     QualificationEntityMapper::class,
   ],
 )
-interface PreviousQualificationsEntityMapper {
+abstract class PreviousQualificationsEntityMapper {
+
+  @Autowired
+  private lateinit var qualificationEntityMapper: QualificationEntityMapper
 
   @ExcludeJpaManagedFieldsIncludingDisplayNameFields
   @GenerateNewReference
   @Mapping(target = "createdAtPrison", source = "prisonId")
   @Mapping(target = "updatedAtPrison", source = "prisonId")
-  fun fromCreateDtoToEntity(dto: CreatePreviousQualificationsDto): PreviousQualificationsEntity
+  abstract fun fromCreateDtoToEntity(dto: CreatePreviousQualificationsDto): PreviousQualificationsEntity
 
   @Mapping(target = "lastUpdatedBy", source = "updatedBy")
   @Mapping(target = "lastUpdatedByDisplayName", source = "updatedByDisplayName")
   @Mapping(target = "lastUpdatedAt", source = "updatedAt")
   @Mapping(target = "lastUpdatedAtPrison", source = "updatedAtPrison")
-  fun fromEntityToDomain(persistedEntity: PreviousQualificationsEntity): PreviousQualifications
+  abstract fun fromEntityToDomain(persistedEntity: PreviousQualificationsEntity): PreviousQualifications
+
+  @ExcludeJpaManagedFieldsIncludingDisplayNameFields
+  @ExcludeReferenceField
+  @Mapping(target = "createdAtPrison", ignore = true)
+  @Mapping(target = "updatedAtPrison", source = "prisonId")
+  @Mapping(target = "qualifications", expression = "java( updateQualifications(entity, dto) )")
+  abstract fun updateEntityFromDto(
+    @MappingTarget entity: PreviousQualificationsEntity,
+    dto: UpdatePreviousQualificationsDto,
+  )
+
+  fun updateQualifications(
+    entity: PreviousQualificationsEntity,
+    dto: UpdatePreviousQualificationsDto,
+  ): List<QualificationEntity> {
+    val existingQualifications = entity.qualifications!!
```

Comment:
I'm a bit concerned about NPEs and am going to do quite a bit of testing here (e.g. with various create/update requests). I suspect there are a few hidden dragons, so I'll try to ensure collections are never null (or we code around that possibility).

I've been playing with the annotation argument `nullValueIterableMappingStrategy = NullValueMappingStrategy.RETURN_DEFAULT` this morning, which will help, but again I'd rather adress this separately with appropriate testing.